packages/flutter_tools/tool/tool_coverage.dart

```diff
@@ -3,165 +3,164 @@
 // found in the LICENSE file.
 
 import 'dart:async';
-import 'dart:convert';
+import 'dart:developer';
 import 'dart:io';
+import 'dart:isolate';
 
-import 'package:args/args.dart';
+import 'package:async/async.dart';
+import 'package:coverage/coverage.dart';
 import 'package:flutter_tools/src/context_runner.dart';
+import 'package:path/path.dart' as p;
+import 'package:stream_channel/isolate_channel.dart';
+import 'package:stream_channel/stream_channel.dart';
+import 'package:test_core/src/runner/hack_register_platform.dart' as hack; // ignore: implementation_imports
+import 'package:test_core/src/executable.dart' as test; // ignore: implementation_imports
+import 'package:vm_service_client/vm_service_client.dart';
+import 'package:test_api/src/backend/runtime.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/suite_platform.dart'; // ignore: implementation_imports
+import 'package:test_core/src/runner/platform.dart'; // ignore: implementation_imports
+import 'package:test_core/src/runner/runner_suite.dart'; // ignore: implementation_imports
+import 'package:test_core/src/runner/suite.dart'; // ignore: implementation_imports
+import 'package:test_core/src/runner/plugin/platform_helpers.dart'; // ignore: implementation_imports
+import 'package:test_core/src/runner/environment.dart'; // ignore: implementation_imports
 import 'package:flutter_tools/src/project.dart';
 import 'package:flutter_tools/src/test/coverage_collector.dart';
-import 'package:pool/pool.dart';
-import 'package:path/path.dart' as path;
-
-final ArgParser argParser = ArgParser()
-  ..addOption('output-html',
-    defaultsTo: 'coverage/report.html',
-    help: 'The output path for the genhtml report.'
-  )
-  ..addOption('output-lcov',
-    defaultsTo: 'coverage/lcov.info',
-    help: 'The output path for the lcov data.'
-  )
-  ..addOption('test-directory',
-    defaultsTo: 'test/',
-    help: 'The path to the test directory.'
-  )
-  ..addOption('packages',
-    defaultsTo: '.packages',
-    help: 'The path to the .packages file.'
-  )
-  ..addOption('genhtml',
-    defaultsTo: 'genhtml',
-    help: 'The genhtml executable.');
-
-
-/// Generates an html coverage report for the flutter_tool.
+
+/// Generates an lcov report for the flutter tool unit tests.
 ///
 /// Example invocation:
 ///
-///    dart tool/tool_coverage.dart --packages=.packages --test-directory=test
+///    dart tool/tool_coverage.dart.
 Future<void> main(List<String> arguments) async {
```

Comment:
The reason for switching to the package:test shim is that it allows us to load the precompiled dill files for testing. This shaves off several seconds of startup time from each test.

Reply:
Is there an issue filed somewhere for doing this through the front door, instead of hacking around?

Reply:
https://github.com/dart-lang/test/issues/36 -  I don't see it being resolved on our timeframe though